environment.c

```diff
@@ -82,9 +82,21 @@ int max_allowed_tree_depth =
 	 * the stack overflow can occur.
 	 */
 	512;
+#else
+#if defined(GIT_WINDOWS_NATIVE) && defined(__clang__) && defined(__aarch64__)
+	/*
+	 * Similar to Visual C, it seems that on Windows/ARM64 the clang-based
+	 * builds have a smaller stack space available. When running out of
+	 * that stack space, a `STATUS_STACK_OVERFLOW` is produced. When the
+	 * Git command was run from an MSYS2 Bash, this unfortunately results
+	 * in an exit code 127. Let's prevent that by lowering the maximal
+	 * tree depth; This value seems to be low enough.
+	 */
+	1280;
```

Comment:
The commit message talks about 1024, but the code uses 1280 as the actual limit.

Reply:
Oh right! 1024 was my first intention (because it's a nice power of two), but then I reconsidered because the gap was _too_ much. 0x600 was too high, so I settled for 0x500 instead. Will adjust the commit message (and rebase to current `main`) once I'm back on a proper computer.

Reply:
Done!